common/persistence/cassandra/cassandraPersistence.go

```diff
@@ -1078,11 +1089,77 @@ func (d *cassandraPersistence) GetShard(
 		}
 	}
 
-	info := createShardInfo(d.currentClusterName, result["shard"].(map[string]interface{}))
+	// check if rangeID column and rangeID field in shard column matches, if not we need to pick the larger
+	// rangeID.
+	//
+	// If shardInfoRangeID < rangeID, we don't need to do anything here as createShardInfo will ignore
+	// shardInfoRangeID and return rangeID instead. Later when updating the shard, CAS can still succeed
+	// as the value from rangeID columns is returned, shardInfoRangeID will also be updated to the correct value.
+	rangeID := result["range_id"].(int64)
+	shard := result["shard"].(map[string]interface{})
+	shardInfoRangeID := shard["range_id"].(int64)
+	if shardInfoRangeID > rangeID {
```

Comment:
If the shard info range ID is deprecated, do we still need this check? 
Should we handle if rangeID > shard info range ID?

Reply:
We will update rangeID field in shardInfo. So we have more chance for catching data corruption. 

I explained the case for rangeID > shard info range ID in the comments on L1095. Basically we don't need to handle that case.